dev/plugins/hu.elte.txtuml.validation/src/hu/elte/txtuml/validation/visitors/DataTypeVisitor.java

```diff
@@ -29,18 +28,15 @@ public boolean visit(FieldDeclaration node) {
 		if (!ElementTypeTeller.isFinal(node)) {
 			collector.report(new MutableDataTypeField(collector.getSourceInfo(), node));
 		}
-		if (!Utils.isBasicType(node.getType(), false)
-				&& !ElementTypeTeller.isDataType(node.getType().resolveBinding())) {
+		if (!Utils.isAllowedAttributeType(node.getType(), false)) {
 			collector.report(new InvalidDataTypeField(collector.getSourceInfo(), node));
 		}
 		return false;
 	}
 
 	@Override
 	public boolean visit(MethodDeclaration node) {
-		if (!node.isConstructor()) {
-			collector.report(new InvalidDataTypeMethod(collector.getSourceInfo(), node));
-		}
+
 		// TODO: check body
```

Comment:
Ok, but as I see currently no validation is supported for method declarations.
Parameter and return type constraints (as written in the Javadoc - can only be decendants of ModelClass, etc. ) are validated somewhere?